### ISSUE TITLE
Release MoRobtrol dataset on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [EMI-Group/evomo#20]
Hi @Zhenyu2Liang 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability.If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your dataset for instance),
you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

Would you like to host the MoRobtrol dataset you've introduced on https://huggingface.co/datasets?
Hosting on Hugging Face will give you more visibility/enable better discoverability, and will also allow people to do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```

If you're down, leaving a guide here: https://huggingface.co/docs/datasets/loading.
We also support Webdataset, useful for image/video datasets: https://huggingface.co/docs/datasets/en/loading#webdataset.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any guidance.

Kind regards,

Niels

**Comment by Zhenyu2Liang** [comment]
Hi @NielsRogge 🤗,

Thank you for your message! I’m happy to submit my paper to the Hugging Face platform to improve its discoverability. I will follow the provided link to submit it.

Regarding the second point about submitting MoRobtrol as a dataset, MoRobtrol is a multiobjective robot control benchmark based on [Brax](https://github.com/google/brax) that requires interaction with the environment and cannot be directly submitted as a static dataset.  However, if you have any suggestions on how to share it effectively, I’d be happy to hear them.

Thanks again for reaching out, and I look forward to contributing further!

Kind regards,  
Zhenyu

**Comment by NielsRogge** [comment]
Ok thanks for the clarifications! I see you have an nn.Module in your README. What you could do is augment it with the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/en/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class so that people can easily push (and load) their trained models to the 🤗 hub.

```diff
+ from huggingface_hub import PyTorchModelHubMixin

+ class SimpleMLP(nn.Module, PyTorchModelHubMixin):
    def __init__(self):
        super(SimpleMLP, self).__init__()
        self.features = nn.Sequential(nn.Linear(8, 4), nn.Tanh(), nn.Linear(4, 2))

    def forward(self, x):
        return torch.tanh(self.features(x))
```
This way, people can do:
```python
model = SimpleMLP()

model.push_to_hub("your-hf-username-or-org/my-trained-mlp")

# reload
model = SimpleMLP.from_pretrained("your-hf-username-or-org/my-trained-mlp")
```

This way, people can easily share their trained models.

**Comment by Zhenyu2Liang** [comment]
Thank you for your suggestion! However, the method we’re using is based on neuroevolution — specifically, we evolve the parameters of MLPs using an evolutionary multiobjective optimization (EMO) algorithm. It’s a gradient-free approach, where users can directly run our algorithm to generate a population of MLP models. These models act as policy networks that continuously interact with the environment to improve their performance.

Because of this setup, it may not be suitable to directly upload individual models to the Hugging Face Hub. That said, we really appreciate your feedback, and if we develop other models that are a better fit for hosting on the Hub, we’ll definitely consider uploading them.

Thanks again for your helpful feedback!

**Comment by NielsRogge** [comment]
Thanks for the clarifications! Looking forward to future releases, happy to help with integrations with the hub.
